src/code/z_camera.c

```diff
@@ -4243,24 +4255,44 @@ s32 Camera_Subj3(Camera* camera) {
     return 1;
 }
 
+/**
+ * Subject 4 uses bgCamData.data differently than other functions:
+ * It uses the only setting where bgCamData.dataCount is either 6 or 9 from the scene files (CAM_SET_CRAWLSPACE)
+ * bgCamData.data[1] stores the coordinates of the front entrance to the crawlspace
+ * bgCamData.data[4] stores the coordinates of the back entrance to the crawlspace
+ * bgCamData.data[0], bgCamData.data[2], bgCamData.data[3], bgCamData.data[5], go unused,
+ *      but also contain coordinates along the straight line path of the crawlspace
+ * Entry refers to the side of the crawlspace first accessible via casual gameplay, Exit is the other entrance
+ */
+#define BGCAM_CRAWLSPACE_FRONT_POS(v) ((v)[1])
+#define BGCAM_CRAWLSPACE_BACK_POS(v, l) ((v)[l - 2])
+#define vCrawlSpaceExitPos temp1
```

Comment:
I copied the example used in OnePointDemo:
https://github.com/zeldaret/oot/blob/d1772c334039a940a187aefa99c82ce1e4b5d002/src/code/z_onepointdemo.c#L1139-L1142
But I like the idea of using #undef if that's a thing, and possibly moving the macros closer to the temps. I think this is important to get right since this will be needed a lot in camera...

Reply:
I don't know how others would see this but yeah you could have
```c
void myBogusFunction() {
    s32 temp;
#define myFavouriteNumber temp
    myFavouriteNumber = 421;
    DoSomething(myFavouriteNumber);
#undef myFavouriteNumber
    DoSomeUnrelatedStuff();
#define yourFavouriteNumber temp
    yourFavouriteNumber = 124;
    ReconsiderLifeChoices(yourFavouriteNumber);
#undef yourFavouriteNumber
    DoSomeUnrelatedStuffAgain();
}
```
the formatter will remove any indentation on preprocessor directives which may be an issue for some people